### PR TITLE
Helm: Publish Helm charts to grafana/helm-charts.

### LINKED ISSUE #6877: [Helm] Publish helm chart to grafana/helm-charts
While we are moving the helm chart source back to this repo, we don't want to change where the chart is hosted. As a result we need to incorporate the CI/GitHub Actions from https://github.com/grafana/helm-charts into this repo to publish the new chart to the same place.

* Migrate CI from grafana/helm-charts for testing helm chart changes
* Integrate the github action in the grafana/helm-charts repo (this will publish it, so we will tackle this last)

### PR DESCRIPTION
**Pull request by jeschkies**
**What this PR does / why we need it**:
This change pulls in the GitHub workflow for publishing Loki Helm charts to `grafana/helm-charts`.

**Which issue(s) this PR fixes**:
Fixes #6877

### CODE REVIEW

File: .github/workflows/helm-release.yaml
Comment (jeschkies) [inline .github/workflows/helm-release.yaml:20]: @trevorwhitney do you know which key we used to use? There's a public key checked into `production/helm`. Should we generate a new key?

File: .github/workflows/helm-release.yaml
Comment (jeschkies) [inline .github/workflows/helm-release.yaml:14]: I wonder whether this will also release `promtail` and `fluent-bit`.

File: .github/workflows/helm-release.yaml
Comment (jeschkies) [inline .github/workflows/helm-release.yaml:8]: Ideally we infer the Loki version. What do you think?

File: .github/workflows/helm-release.yaml
Comment (trevorwhitney) [reply .github/workflows/helm-release.yaml:20]: yikes, no, I have no idea. is there a way we can reverse engineer what key was being used in the `grafana/helm-charts` repo? maybe we could ask @krajorama and @Logiraptor what they did when the moved the mimir helm chart?

File: .github/workflows/helm-release.yaml
Comment (trevorwhitney) [reply .github/workflows/helm-release.yaml:14]: yes, my guess is it will. I think we could either a) remove everything but the README from those charts, and include the SHA they were deleted at (in a separate PR that we merge before this one), or b) can we sepecify a `charts_dir` of `production/helm/loki`?

I actually prefer option `a` as I think it cleans up the code base a bit.

File: .github/workflows/helm-release.yaml
Comment (trevorwhitney) [reply .github/workflows/helm-release.yaml:8]: What do you mean by that, not sure I'm following?

**Review by trevorwhitney — CHANGES_REQUESTED** [review]
This looks great, thanks for adding this. I'm going to put a request changes on this just as a precaution as once this is merged the chart will publish, so I want to make sure we're good to go before that.

File: .github/workflows/helm-release.yaml
Comment (trevorwhitney) [reply .github/workflows/helm-release.yaml:14]: https://github.com/grafana/loki/pull/6978 <- PR removing the deprecated helm charts

File: .github/workflows/helm-release.yaml
Comment (Logiraptor) [reply .github/workflows/helm-release.yaml:20]: I'll have to defer to @krajorama on what was done. I can see a signing key exists in the settings of the mimir repo, but I can't view the secret contents.

File: .github/workflows/helm-release.yaml
Comment (jeschkies) [reply .github/workflows/helm-release.yaml:8]: I meant we infer the Loki version by the tag for this flow. This is not really connected to this line though.

**Review by trevorwhitney — APPROVED** [review]
LGTM

File: .github/workflows/helm-release.yaml
Comment (jeschkies) [reply .github/workflows/helm-release.yaml:20]: We are still missing the key.

File: .github/workflows/helm-release.yaml
Comment (slim-bean) [reply .github/workflows/helm-release.yaml:8]: I don't think we want this workflow to run on tags, we typically push the tag for the release in order to build the binaries and images, and _then_ we update the helm charts/jsonnet in a separate step only after the release artifacts exist.

Comment (trevorwhitney) [reply .github/workflows/helm-release.yaml:8]: I agree, let's remove the `tags` section alltogehter, which also more closely matches what mimir does:
```yaml  
push:
    branches:
      - main
      - "release-[0-9]+.[0-9]+"
```

Note: we don't wan the `release-` bit either. They've done some work to add releasing the helm chart to their release workflow, but we're not there yet.

**Pull request merged**
